src/widgets/FormControl.js

```diff
@@ -229,54 +222,102 @@ const FormControlComponent = ({
     [SaveButton, CancelButton]
   );
 
-  const InputConfirm = (
-    <ConfirmForm
-      isOpen={isConfirmFormOpen}
-      questionText={confirmText}
-      onConfirm={onSaveCompletedForm}
-      confirmText={modalStrings.confirm}
-    />
-  );
-
-  const InputDetails = (
-    <View style={localStyles.flexOne}>
-      <ScrollView style={localStyles.whiteBackground}>
-        <View style={localStyles.flexRow}>
-          <View style={localStyles.flexOne} />
-          <View style={localStyles.flexTen}>{formInputs()}</View>
-          <View style={localStyles.flexOne} />
-        </View>
-      </ScrollView>
-      <Buttons />
-    </View>
+  return (
+    <>
+      <View
+        style={
+          isConfirmFormOpen
+            ? { ...localStyles.flexOne, ...localStyles.hidden }
+            : localStyles.flexOne
+        }
+      >
+        <ScrollView style={localStyles.whiteBackground}>
+          <View style={localStyles.flexRow}>
+            <View style={localStyles.flexOne} />
+            <View style={localStyles.flexTen}>{formInputs()}</View>
+            <View style={localStyles.flexOne} />
+          </View>
+        </ScrollView>
+        <Buttons />
+      </View>
+      <View
+        style={
+          isConfirmFormOpen
+            ? localStyles.flexOne
+            : { ...localStyles.flexOne, ...localStyles.hidden }
+        }
+      >
+        <ConfirmForm
+          isOpen={isConfirmFormOpen}
+          questionText={confirmText}
+          onConfirm={onSaveForm}
+          onCancel={onCancelForm}
+          confirmText={modalStrings.confirm}
+        />
+      </View>
+    </>
   );
+};
 
-  return isConfirmFormOpen ? InputConfirm : InputDetails;
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
```

Comment:
![giphy](https://user-images.githubusercontent.com/43223637/79739000-aad69280-8351-11ea-91a3-bb9fcb85ce04.gif)

Reply:
I think it just means code for the component is harder to read, looking in quite a few different places. Think it would be good (not now) idea to use only `mergeProps` or `mapDispatchToProps` - the dispatch logic is just too scattered and hard to read

Reply:
not sure that I follow you @joshxg - I thought the `mergeProps` was ok here. makes the wiring section at the bottom of the component much more involved, sure, but doesn't it mean that the declarative part of the component is slightly neater?

Reply:
ya the declarative part is way better and I do like the use of `mergeProps`, not saying to change - just it was a little harder to read dispatch code when it was over two different functions. If you didn't use a `mapDispatchToProps`, `dispatchProps` would be just the `dispatch` function in `mergeProps`, so could just have all the code there - a bit easier to scan maybe (but at that point maybe you should just return state from `mapStateToProps` and have all of it in `mergeProps` 🤷!

Reply:
@joshxg Yup, that was my thought, too. But my thinking was that it's actually tidier to have all the selectors in the `mapState`, all the dispatches in `mapDispatch` and then all the conditional logic in `mergeProps`. With components with more complex prop logic, I think it's actually tidier to split it up like that and isn't really that much of a difficult pattern to follow if it's applied consistently?

Reply:
I'm happy either way, it's really just styling which really we're all happy with! Last point but leaving it here 🤣 

I believe `mergeProps` return isn't cached, where as `mapDispatchToProps` is, so the new callbacks will always have new references. So you come back to using (when needed!) to `useCallback`, anyway

Reply:
got it, thanks for clarifying.
I rather like the split of having `mapDispatch` handle callbacks and `mergeProps` handle the conditional logic. Though if it isn't cached there is another problem and more code. as long as this pattern is an edge case for us, only when necessary..